src/content/drafts/custom-formatter-blog-post.md

```diff
@@ -0,0 +1,427 @@
+line = 2
+message = "Expected { after 'if' condition."
+nodeType = "IfStatement"
+ruleId = "curly"
+severity = 2
+[[0.messages]]
+column = 1
+line = 3
+message = "Don't use process.exit(); throw an error instead."
+nodeType = "CallExpression"
+ruleId = "no-process-exit"
+severity = 2
+source = "var err = doStuff();\nif (err) console.log('failed tests: ' + err);\nprocess.exit(1);\n"
+warningCount = 0
+[1]
+errorCount = 0
+filePath = "/path/to/Gruntfile.js"
+fixableErrorCount = 0
+fixableWarningCount = 0
+messages = []
+warningCount = 0
+```
+
+You'll also see how to publish the formatter to npm and use it in a project.
+
+## Steps
+
+Before you begin:
+
+1. Have [Node.js](https://nodejs.org) and npm installed in your development environment.
+2. Have a basic understanding of how to use Node.js and npm.
+3. If you want to publish your custom formatter to npm like you'll cover below, 
+   [create an npm account](https://www.npmjs.com/signup) and login from the
+   [npm CLI](https://docs.npmjs.com/cli/v7/commands/npm-adduser).
+
+### 1. Create a project
+
+First, you're going to create the project for your custom formatter.
+Since an ESLint custom formatter is just a JavaScript function, create a 
+new project for the custom formatter by running the following:
+
+```bash
+mkdir eslint-formatter-toml
+cd eslint-formatter-toml
+npm init -y
+```
+
+Add ESLint to the project:
+
+```bash
+npm init -y @eslint/config
+```
+
+Select the following options when prompted:
+
+| Prompt | Answer |
+|:-------|:-------|
+| How would you like to use ESLint? | To check syntax and find problems |
+| What type of modules does your project use? | CommonJS (require/exports) | 
+| Which framework does your project use? | None of these |
+| Does your project use TypeScript? | No |
+| Where does your code run? | Node |
+| What format do you want your config file to be in? | JavaScript |
+
+This creates the file `.eslintrc.js` with your ESLint configuration.
+The file should look like:
+
+```js
+module.exports = {
+    "env": {
+        "browser": true,
+        "commonjs": true,
+        "es2021": true,
+    },
+    "extends": "eslint:recommended",
+    "overrides": [
+    ],
+    "parserOptions": {
+        "ecmaVersion": "latest"
+    },
+    "rules": {
+    }
+}
+```
+
+For more information on customizing your ESLint configuration,
+refer to the [Configuration File documentation](https://eslint.org/docs/latest/user-guide/configuring/configuration-files).
+
+Next, add the [json2toml](https://www.npmjs.com/package/json2toml) package, which you'll use to help convert JavaScript objects to TOML: 
+
+```bash
+npm install json2toml
+```
+
+Now you're ready to write the custom formatter.
+
+### 2. Create the custom formatter 
+
+Custom formatters are just JavaScript functions that receive a `results` object and 
+optional `context` object as arguments and return a string as an output.
+In the custom formatter you'll make in this post, you'll just use the `results` argument.
```

Comment:
Optional suggestion: I'm cautious about using "just" when teaching. I'll assume you're familiar with that guideline, and I'm happy to elaborate if not. There are a couple instances of "just" as in "just a function" prior to this where conveying that something is simple is probably be the intention. Here's a case where we can substitute the more specific "only" with no loss of clarity:

```suggestion
In the custom formatter you'll make in this post, you'll only use the `results` argument.
```

Or here's a slightly rearranged variation:

```suggestion
The custom formatter you'll make in this post will only use the `results` argument.
```

Reply:
i don't think i had been explicitly familiar with the "do not use just" as a guideline, although I definitely have also suggested that writers not include it while performing copy reviews. 

after a [bit of reading](https://www.excitant.co.uk/just-is-a-dangerous-word/) makes sense to me as a guideline.  

thanks for catching this and sharing!

Reply:
How fortuitous, then! I'm glad there was an opportunity to share.